### PR TITLE
Set name of current route as transaction

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
This change adds the current route as `SentryEvent.transaction`.

## :bulb: Motivation and Context
https://develop.sentry.dev/sdk/event-payloads/ suggests to use the current route as transaction name.

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/navigation/sentry_navigator_observer.dart [hunk 1]
```diff
@@ -35,13 +35,19 @@ const _navigationKey = 'navigation';
...
 class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
-  SentryNavigatorObserver({Hub? hub}) : hub = hub ?? HubAdapter();
+  SentryNavigatorObserver({Hub? hub, this.setRouteNameAsTransaction = true})
+      : _hub = hub ?? HubAdapter();
 
-  final Hub hub;
+  final Hub _hub;
+
+  /// Enabling this option overrides the current [Scope.transaction].
+  /// So be careful when this is used together with performance tracing.
```
Comment (marandaneto) [inline flutter/lib/src/navigation/sentry_navigator_observer.dart]:
```suggestion
  /// Enabling this option overrides the current [Scope.transaction].
  /// So be careful when this is used together with performance monitoring.
```

File: flutter/lib/src/navigation/sentry_navigator_observer.dart [hunk 2]
```diff
@@ -35,13 +35,19 @@ const _navigationKey = 'navigation';
...
+  final bool setRouteNameAsTransaction;
```
Comment (marandaneto) [inline flutter/lib/src/navigation/sentry_navigator_observer.dart]:
```suggestion
  final bool _setRouteNameAsTransaction;
```

Comment (marandaneto) [reply flutter/lib/src/navigation/sentry_navigator_observer.dart]: can we make it private?

**Review by marandaneto — APPROVED** [review]
left a comment other than that, LGTM

File: flutter/lib/src/navigation/sentry_navigator_observer.dart (hunk 2 above)
Comment (ueman) [reply flutter/lib/src/navigation/sentry_navigator_observer.dart]: No, because it's a constructor param. But because it's a final property it's read only.

Comment (marandaneto) [reply flutter/lib/src/navigation/sentry_navigator_observer.dart]: I mean, we can do the same as `_hub` and `hub`, so we dont expose the `setRouteNameAsTransaction`

**Pull request merged**